src/Components/EventDetail.jsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect } from "react"
+import { useParams } from "react-router";
+import { fetchEvent } from "../app/fetchEvent"
+import { setEventDetail } from "../features/event/eventSlice"
+import { useDispatch, useSelector } from "react-redux";
+import { Card, Row, Col, Avatar, Tooltip, Space, Image } from "antd";
+import { IconText } from "../features/functions/IconText";
+import { StarOutlined } from "@ant-design/icons";
+export default function EventDetail() {
+
+    const { id } = useParams();
+    const eventDetail = useSelector(state => state.event.eventDetail);
+    const dispatch = useDispatch();
+
+    const gridCenter = {
+        width: '100%',
+        textAlign: 'center',
+    };
+    const gridLeft = {
+        width: '100%',
+        textAlign: 'left',
+    };
+    useEffect(() => dispatch(fetchEvent(id, setEventDetail)),
+        [id, dispatch]);
+
+    if (eventDetail === null) {
+        return <div>Hello</div>
+    }
+  
+    const rate = eventDetail.reviews === null ? "NaN" : eventDetail.reviews.reduce((prev, cur) => prev + cur.rate, 0) / eventDetail.reviews.length;
```

Comment:
Can abstract the reduce into its own function so you can reuse it. Also, you can use `useMemo` here to optimize rendering performance.